src/server/player_sao.cpp

```diff
@@ -574,28 +574,9 @@ bool PlayerSAO::checkMovementCheat()
 		m_last_good_position = m_base_position;
 		return false;
 	}
-	if (UnitSAO *parent = dynamic_cast<UnitSAO *>(getParent())) {
-		v3f attachment_pos;
-		{
-			int parent_id;
-			std::string bone;
-			v3f attachment_rot;
-			bool force_visible;
-			getAttachment(&parent_id, &bone, &attachment_pos, &attachment_rot, &force_visible);
-		}
-
-		v3f parent_pos = parent->getBasePosition();
-		f32 diff = m_base_position.getDistanceFromSQ(parent_pos) - attachment_pos.getLengthSQ();
-		const f32 maxdiff = 4.0f * BS; // fair trade-off value for various latencies
 
-		if (diff > maxdiff * maxdiff) {
-			setBasePosition(parent_pos);
-			actionstream << "Server: " << m_player->getName()
-					<< " moved away from parent; diff=" << sqrtf(diff) / BS
-					<< " resetting position." << std::endl;
-			return true;
-		}
-		// Player movement is locked to the entity. Skip further checks
+	if (getParent()) {
```

Comment:
This should be merged with the previous condition so that `m_last_good_position` is updated too

Reply:
It might also be better to have `FATAL_ERROR_IF(getParent(), "...")` since checkMovementCheat shouldn't run anyway when attached.

Reply:
It should also not be run in singleplayer and when anticheat is disabled. I think these checks should belong into `checkMovementCheat` for consistency in case the function is called elsewhere too (code duplication).